test/test_images/sendevents/main.go

```diff
@@ -47,6 +49,8 @@ func init() {
 	flag.StringVar(&eventID, "event-id", "", "Event ID to use. Defaults to a generated UUID")
 	flag.StringVar(&eventType, "event-type", "knative.eventing.test.e2e", "The Event Type to use.")
 	flag.StringVar(&eventSource, "event-source", "", "Source URI to use. Defaults to the current machine's hostname")
+	flag.StringVar(&eventExtensionName, "event-extension-name", "", "The extension name of event.")
```

Comment:
as talked offline, we can receive a `eventExtensions` parameter, which is a comma separated value of extensions names and values... E.g., `extensionA=extensionAValue,extensionB=extensionBValue`

Reply:
`json` sounds more reasonable and easy-to-parse for me :-)

Reply:
By using `json` format you can convert it to `Extensions` map simply with `json.Unmarshal`, like https://github.com/knative/eventing/blob/1b5a2b94ccdddd5274fda1a91abf943a10a198ae/test/test_images/sendevents/main.go#L123